### DISCUSSION TITLE
Interpreting CLV Frequency-Recency heatmap

### DISCUSSION THREAD

**Discussion by AMeynen** [pymc-labs/pymc-marketing#356] [Q&A] [answered]
I need some help in the interpretation of this Frequency/Recency graphs of the **same** beta-geometric model.

![image](https://github.com/pymc-labs/pymc-marketing/assets/40566928/543fe3dd-b8d1-4639-921a-2de3f50fb08c)
The left plot is a plot with max_recency = 365, the right plot is a plot with max_recency = 730. The documentation states the following about the max_recency parameter: max_recency: The maximum recency to plot. This also determines the age of the customer. Default to max observed age.

I find it striking that the expected number of future purchases for the same recency-frequency differs between the two graphs. E.g. Recency = 350, Frequency = 50 -> 4 on first graph, 0 on second graph.

My current interpretation is that this plot shows the expect number of purchases if the maximum observed recency was equal to max_recency. Could someone confirm whether my interpretation is correct?

The reason why I have to limit the max_recency and max_frequency is due to memory issues. Solutions for this might be helpful as well.

KR
Alex

**Comment by ColtAllen** [comment, marked as answer]
Hey @AMeynen,

Was this model fit with the default MCMC? If so, then `chain` X `draw` computations are probably being made for every value in the plotted matrix, which could explain the memory issues. Try plotting with `model.fit(fit_method = 'map')` and let us know what happens.

> My current interpretation is that this plot shows the expect number of purchases if the maximum observed recency was equal to max_recency. Could someone confirm whether my interpretation is correct?

Your interpretation is correct. This plotting function also sets `T=max_recency`, which is why the results appear so different between the plots. `T` is an important parameter, and assuming every customer is the same age exposes a fundamental limitation in this plot, so thanks for bringing it to our attention. It may be better if this plot were 3-dimensional.
